### PR TITLE
Solves & Fixes Issue No 10488 | Redundant constant term removed from Integral Value

### LINKED ISSUE #10488: integrate(x/(a*x+b), x) gives wrong answer
Using sympy version 0.7.5, for the following code:

```
import sympy
x, a, b  = sympy.symbols('x,a,b')
print(sympy.integrate(x/(a*x+b), x))
```
I get:

`x/a - b*log(a**2*x + a*b)/a**2`

which is wrong.  I should get:
`x/a - b*log(a*x + b)/a**2`

I'm not sure where the extra 'a' multiple within the log comes from.

Same thing happens on http://live.sympy.org/ which is sympy version 0.7.6

### PR DESCRIPTION
**Pull request by ashutoshsaboo**
Hello.

Fixes and Solves #10488 . Support for h.primitive() added, to remove the redundant terms from the integral value. 

This issue is based on the output of this code-:

```
import sympy
x, a, b  = sympy.symbols('x,a,b')
print(sympy.integrate(x/(a*x+b), x))
```

Previously, this gave an output as-:

`x/a - b*log(a**2*x + a*b)/a**2`

Now, after the new code, this is the new output-:

`x/a - b*log(a*x + b)/a**2`

Hence, as can be observed, the redundant term of b*log(a) is removed, and hence this fixes #10488 .

@asmeurer @jksuom @rtrwalker

### CODE REVIEW

File: sympy/integrals/rationaltools.py [hunk 1]
```diff
@@ -8,7 +8,7 @@
 from sympy.polys import Poly, resultant, ZZ
 from sympy.polys.polytools import count_roots
 from sympy.core.compatibility import range
-
+from sympy import primitive
```
Comment (jksuom) [inline sympy/integrals/rationaltools.py]: There should be no need for this import.

File: sympy/integrals/rationaltools.py [hunk 2]
```diff
@@ -92,17 +92,19 @@ def ratint(f, x, **flags):
 
         if not real:
             for h, q in L:
+                h=h.primitive()
```
Comment (jksuom) [inline sympy/integrals/rationaltools.py]: ``h,primitive()`` returns a tuple. Hence it is more logical to set ``_, h = h.primitive()`` (the coefficient in the first place is not needed; therefore ``_``.)

File: sympy/integrals/rationaltools.py [hunk 3]
```diff
@@ -92,17 +92,19 @@ def ratint(f, x, **flags):
...
                 eps += RootSum(
-                    q, Lambda(t, t*log(h.as_expr())), quadratic=True)
+                    q, Lambda(t, t*log(h[1].as_expr())), quadratic=True)
```
Comment (jksuom) [inline sympy/integrals/rationaltools.py]: This could be left as is.

**Comment by ashutoshsaboo** [comment]
@jksuom Thanks! I have pushed a new commit, with the required changes suggested by you. 

Any more changes in the code that you'd suggest? Or can this get merged now? :smiley:

**Comment by jksuom** [comment]
It looks good to me. Just a minor point to keep in mind in the future: It is recommended that a space be added on either side of an assignment (``=``) ([Other Recommendations](https://www.python.org/dev/peps/pep-0008/#id25)).

**Comment by ashutoshsaboo** [comment]
@jksuom Ohh I see. I'll remember that from next time onwards. Should I push a new commit with those spaces?

**Comment by jksuom** [comment]
I don't think that will be necessary. A new commit would create a new Travis build job that would be placed at the end of the queue that is quite long even now.

**Comment by ashutoshsaboo** [comment]
@jksuom Okay! Sure! Thanks! :+1:

**Comment by ashutoshsaboo** [comment]
@jksuom Some of the Travis CI Build Tests failed. I tried to look into them, but I am not able to understand, because the only difference that we did was the - `h=h.primitive()`.

Could you help me out, to pass those tests? @jksuom

**Comment by jksuom** [comment]
It seems that the algorithm applies integration to find the generators for the Lie group leaving a differential equation invariant. Such generators are not unique; any constant multiple will do as well. The result of the failing test was affected by the change in ratint. The original result contains
``xi(x, f(x)): sqrt(2*a0 + 2*a1*x + 2*a2*x**2 + 2*a3*x**3 + 2*a4*x**4)}``
from which the constant factor ``sqrt(2)`` can be removed to simplify the result. The change in ratint does this. So you should edit this expression in sympy/solvers/tests/test_ode.py, line 2394,  by removing all multipliers ``2*`` in the argument of ``sqrt`` (so it becomes ``sqrt(a0 + a1*x + ...)``).

Before committing the change you should test it locally by running ``bin/test test_ode``.

(This is also a suitable time to add the spaces on either side of ``=``.)

**Comment by ashutoshsaboo** [comment]
@jksuom Hmm! I made the change, and ran the test, and it didn't give any such errors. I have pushed the new commit. Also, I have added the spaces, as you suggested.

You can review it, and advise me of any other change in the code? 

Thanks! :smile: @jksuom @asmeurer

**Comment by jksuom** [comment]
Looks good. I think this was the only place to be adjusted. Let's see.

File: sympy/solvers/tests/test_ode.py [hunk 4]
```diff
@@ -2391,7 +2391,7 @@ def test_heuristic1():
     i4 = infinitesimals(eq4, hint='abaco1_simple')
     assert i4 == [{eta(x, f(x)): 1, xi(x, f(x)): 0},
         {eta(x, f(x)): 0,
-        xi(x, f(x)): sqrt(2*a0 + 2*a1*x + 2*a2*x**2 + 2*a3*x**3 + 2*a4*x**4)}]
```
Comment (asmeurer) [inline sympy/solvers/tests/test_ode.py]: What does this represent? It would be good to double check that this answer is still mathematically correct.

Comment (asmeurer) [inline sympy/solvers/tests/test_ode.py]: Oh I see this was already addressed by the other comments here.

**Comment by asmeurer** [comment]
+1. I like this very simple fix. We could probably apply a similar fix to risch.py as well.

**Comment by ashutoshsaboo** [comment]
Hi! @jksuom Hmm! So the last commit passed all the Travis CI Build Tests. Can we get this merged now?

@asmeurer Is that for some other issue or what is it? Could you elaborate?

Thanks! :smiley:

**Comment by jksuom** [comment]
Yes, I think this is ready. Thanks!

**Pull request merged**